File: mobile/ios/shotclock/ShotClock/Models/Session.swift
import Foundation

enum SessionStatus: String, Codable {
    case lobby, active, paused, ended
}

struct PowerHourSession: Codable, Identifiable {
    let id: String
    let admin: Int
    let title: String
    let inviteCode: String

    // Configuration
    let tracksPerPlayer: Int
    let maxTracks: Int
    let secondsPerTrack: Int
    let transitionClip: String
    let hideTrackOwners: Bool

    // State
    let status: SessionStatus
    let currentTrackIndex: Int

    let createdAt: String
    let startedAt: String?
    let endedAt: String?

    // Included in list responses
    let playerCount: Int?
    let trackCount: Int?

    var isActive: Bool {
        status == .active || status == .paused
    }

    var statusLabel: String {
        switch status {
        case .lobby: return "Lobby"
        case .active: return "Playing"
        case .paused: return "Paused"
        case .ended: return "Ended"
        }
    }
}

struct CreateSessionRequest: Encodable {
    let title: String
    let tracksPerPlayer: Int
    let maxTracks: Int
    let secondsPerTrack: Int
    let transitionClip: String
    let hideTrackOwners: Bool
}

struct JoinSessionRequest: Encodable {
    let inviteCode: String
}

File: mobile/android/shotclock/app/src/main/java/fm/shotclock/mobile/model/Session.kt
package fm.shotclock.mobile.model

import fm.shotclock.mobile.data.network.dto.SessionDto

enum class SessionStatus(val label: String) {
    LOBBY("Lobby"),
    ACTIVE("Active"),
    PAUSED("Paused"),
    ENDED("Ended"),
    ;

    companion object {
        fun fromString(value: String): SessionStatus = when (value.lowercase()) {
            "lobby" -> LOBBY
            "active" -> ACTIVE
            "paused" -> PAUSED
            "ended" -> ENDED
            else -> LOBBY
        }
    }
}

data class PowerHourSession(
    val id: String,
    val title: String,
    val inviteCode: String,
    val status: SessionStatus,
    val tracksPerPlayer: Int,
    val maxTracks: Int,
    val secondsPerTrack: Int,
    val transitionClip: String,
    val hideTrackOwners: Boolean,
    val currentTrackIndex: Int,
    val adminId: Int,
    val playerCount: Int,
    val trackCount: Int,
    val createdAt: String?,
    val startedAt: String?,
    val endedAt: String?,
)

fun SessionDto.toDomain(): PowerHourSession = PowerHourSession(
    id = id,
    title = title,
    inviteCode = inviteCode,
    status = SessionStatus.fromString(status),
    tracksPerPlayer = tracksPerPlayer,
    maxTracks = maxTracks,
    secondsPerTrack = secondsPerTrack,
    transitionClip = transitionClip,
    hideTrackOwners = hideTrackOwners,
    currentTrackIndex = currentTrackIndex,
    adminId = admin ?: 0,
    playerCount = playerCount,
    trackCount = trackCount,
    createdAt = createdAt,
    startedAt = startedAt,
    endedAt = endedAt,
)
